promql/testdata/native_histograms.test

```diff
@@ -0,0 +1,207 @@
+	{} 0
+
+eval instant at 5m histogram_fraction(-Inf, +Inf, empty_histogram{})
+	{} NaN
+
+eval instant at 5m histogram_fraction(0, 8, empty_histogram{})
+	{} NaN
+
+
+
+# buckets:[1 2 1] means 1 observation in the 1st bucket, 2 observations in the 2nd and 1 observation in the 3rd (total 4).
+load 5m
+	single_histogram{}	{{schema:0 sum:4 count:4 buckets:[1 2 1]}}
+
+# histogram_count extracts the count property from the histogram.
+eval instant at 5m histogram_count(single_histogram{})
+	{} 4
+
+# histogram_sum extracts the sum property from the histogram.
+eval instant at 5m histogram_sum(single_histogram{})
+	{} 4
+
+# We expect half of the values to fall in the range 1 < x <= 2.
+eval instant at 5m histogram_fraction(1, 2, single_histogram{})
+	{} 0.5
+
+# We expect all values to fall in the range 0 < x <= 8.
+eval instant at 5m histogram_fraction(0, 8, single_histogram{})
+	{} 1
+
+# Median is 1.5 due to linear estimation of the midpoint of the middle bucket, whose values are within range 1 < x <= 2.
+eval instant at 5m histogram_quantile(0.5, single_histogram{})
+	{} 1.5
+
+
+
+# Repeat the same histogram 10 times.
+load 5m
+	multi_histogram{}	{{schema:0 sum:4 count:4 buckets:[1 2 1]}}x10
+
+eval instant at 5m histogram_count(multi_histogram{})
+	{} 4
+
+eval instant at 5m histogram_sum(multi_histogram{})
+	{} 4
+
+eval instant at 5m histogram_fraction(1, 2, multi_histogram{})
+	{} 0.5
+
+eval instant at 5m histogram_quantile(0.5, multi_histogram{})
+	{} 1.5
+
+
+# Each entry should look the same as the first.
+eval instant at 50m histogram_count(multi_histogram{})
+	{} 4
+
+eval instant at 50m histogram_sum(multi_histogram{})
+	{} 4
+
+eval instant at 50m histogram_fraction(1, 2, multi_histogram{})
+	{} 0.5
+
+eval instant at 50m histogram_quantile(0.5, multi_histogram{})
+	{} 1.5
+
+
+
+# Accumulate the histogram addition for 10 iterations, offset is a bucket position where offset:0 is always the bucket
+# with an upper limit of 1 and offset:1 is the bucket which follows to the right. Negative offsets represent bucket
+# positions for upper limits <1 (tending toward zero), where offset:-1 is the bucket to the left of offset:0.
+load 5m
+	incr_histogram{}	{{schema:0 sum:4 count:4 buckets:[1 2 1]}}+{{sum:2 count:1 buckets:[1] offset:1}}x10
+
+eval instant at 5m histogram_count(incr_histogram{})
+	{} 5
+
+eval instant at 5m histogram_sum(incr_histogram{})
+	{} 6
+
+# We expect 3/5ths of the values to fall in the range 1 < x <= 2.
+eval instant at 5m histogram_fraction(1, 2, incr_histogram{})
+	{} 0.6
+
+eval instant at 5m histogram_quantile(0.5, incr_histogram{})
+	{} 1.5
+
+
+eval instant at 50m histogram_count(incr_histogram{})
+	{} 14
+
+eval instant at 50m histogram_sum(incr_histogram{})
+	{} 24
+
+# We expect 12/14ths of the values to fall in the range 1 < x <= 2.
+eval instant at 50m histogram_fraction(1, 2, incr_histogram{})
+	{} 0.8571428571428571
+
+eval instant at 50m histogram_quantile(0.5, incr_histogram{})
+	{} 1.5
```

Comment:
I would also test the full histogram (i.e. `eval instant at 50m incr_histogram`).

Reply:
Added in 3eb0735a7, see above regarding parser support.